plugins/modules/vmware_drs_group_manager.py

```diff
@@ -0,0 +1,547 @@
+                    need_reconfigure = True
+
+            group_spec = vim.cluster.GroupSpec(info=group, operation='edit')
+            config_spec = vim.cluster.ConfigSpecEx(groupSpec=[group_spec])
+
+            # Check if dry run
+            if not self.module.check_mode and need_reconfigure:
+                task = self._cluster_obj.ReconfigureEx(config_spec, modify=True)
+                self._changed, dummy = wait_for_task(task)
+
+            self._set_result(group)
+            if self._changed:
+                self._msg = "Updated vm group %s successfully" % self._group_name
+            else:
+                self._msg = "No update to vm group %s" % self._group_name
+        else:
+            self._changed = False
+            self._msg = "No update to vm group %s" % self._group_name
+
+    def _normalize_group_data(self, group_obj):
+        """
+        Return human readable group spec
+        Args:
+            group_obj: Group object
+
+        Returns: DRS group object fact
+
+        """
+        if not all([group_obj]):
+            return {}
+
+        # Check if group is a host group
+        if hasattr(group_obj, 'host'):
+            return dict(
+                group_name=group_obj.name,
+                hosts=self._host_list,
+                type="host"
+            )
+        return dict(
+            group_name=group_obj.name,
+            vms=self._vm_list,
+            type="vm"
+        )
+
+    def manage_drs_group_members(self):
+        """
+        Add a DRS host/vm group members
+        """
+
+        if self._vm_list is None:
+            self._manage_host_group()
+        elif self._host_list is None:
+            self._manage_vm_group()
+        else:
+            raise Exception('Failed, no hosts or vms defined')
+
+
+def main():
+    """
+    Main function
+    """
+
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        datacenter=dict(type='str', required=False, aliases=['datacenter_name']),
+        cluster=dict(type='str', required=True, aliases=['cluster_name']),
+        group_name=dict(type='str', required=True),
+        vms=dict(type='list', elements='str'),
+        hosts=dict(type='list', elements='str')
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[['vms', 'hosts']],
+        required_one_of=[['vms', 'hosts']]
+    )
+
+    try:
+        # Create instance of VmwareDrsGroupMemberManager
+        vmware_drs_group = VmwareDrsGroupMemberManager(module=module)
+        vmware_drs_group.manage_drs_group_members()
+
+        # Set results
+        results = dict(msg=vmware_drs_group.get_msg(),
+                       failed=False,
+                       changed=vmware_drs_group.get_changed(),
+                       drs_group_member_info=vmware_drs_group.get_result())
```

Comment:
I don't understand the need for `get_msg()`, `get_changed()` and `get_result()`. I think they make this module unnecessarily complicated.

I know that getters and setters are OOP orthodoxy, but I never understood why... in 99.9% of all cases I've seen so far, they just let you access some member variable directly without any checks or transformations. In other words, you don't have any advantage but you have the disadvantage of unnecessary methods.

While this doesn't make sense in languages like Java (where you can have private variables) it's even more useless in Python where a leading `_` is a kind of hint that this is an internal variable and shouldn't be used directly... without preventing you to do so.

Well, that's my opinion. Maybe I'm wrong and I'm just just too stupid to understand the benefits of getters and setters.